consensus/propagation/reactor.go

```diff
@@ -29,6 +28,11 @@ const (
 
 	// WantChannel the propagation reactor channel handling the wants.
 	WantChannel = byte(0x51)
+
+	// blockCacheSize determines the number of blocks to keep in the cache.
+	// After each block is committed, only the last `blockCacheSize` blocks are
+	// kept.
+	blockCacheSize = 5
```

Comment:
IMO, this should be increased. Assuming 128mb blocks, we can assume this will hold 800mb of RAM (including proofs and other stuff), which is still small.

The issue is the blocktime ~3s. this only keeps the last 15sec compact blocks. If a node falls behind for more than 15sec (which I think is a lot likely to happen), will block sync be started automatically? I don't think that's the case, and this will leave the node hanging if all peers prune.

Maybe increase it to `25`, which would keep ~4GB of RAM but that's fine IMO

Reply:
agree that this number will likely have to be changed in the future, however block parts can just be loaded from the store.